src/pk_core.c

```diff
@@ -355,6 +355,10 @@ static inline bool validateIndex(PKVM* vm, int64_t index, uint32_t size,
   return true;
 }
 
+bool isAByte(int64_t num) {
+  return (num >= -128) && (num <= 127);
+}
+
```

Comment:
The signedness of a char is "implementation defined" (according to the C standards), use `CHAR_MIN` and `CHAR_MAX` instead.

Also, I don't want this to be another function, we can just inline this, or if you prefer use a macro.
```c
#define IS_NUM_BYTE(num) ((CHAR_MIN <= (num)) && ((num) <= CHAR_MAX))
```